flink-core/src/main/java/org/apache/flink/api/common/operators/CollectionExecutor.java

```diff
@@ -192,14 +193,16 @@ private void initCache(Set<Map.Entry<String, DistributedCache.DistributedCacheEn
                                     executionConfig,
                                     cachedFiles,
                                     accumulators,
-                                    metrics)
+                                    metrics,
+                                    new JobID())
```

Comment:
I am wonder wether we could resolve a more global job id from followings in order:
1. {{Plan.getJobId}} if not null.
2. Fallback to a job id for {{CollectionExecutor.execute}} scope.

I think it is helpful and meaningful  for logs. It is also more aligned with other environments.

Reply:
Good point! 
Let me adjust the PR to address this.